pkg/virtctl/set/set_test.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Contributors
+ *
+ */
+
+package set_test
+
+import (
+	"context"
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/mock/gomock"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests/clientcmd"
+)
+
+var _ = Describe("Set command", func() {
+	var vmInterface *kubecli.MockVirtualMachineInterface
+	var ctrl *gomock.Controller
+	const vmName = "testvm"
+	const expectedCpuSocketsCount = 2
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).AnyTimes()
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Context("Input parameters validation", func() {
+		It("should fail with missing input parameters", func() {
```

Comment:
This can be make more compact with a DescribeTable
```golang
DescribeTable("Input parameters validation",func(flags []string, errMsg string){
        cmd := clientcmd.NewRepeatableVirtctlCommand("set", vmName, flags...)
        Expect(cmd()).To(MatchError(errMsg))
},
        Entry("with missing input parameters",[]string{}, "argument validation failed"),
       ....
)
```

Reply:
thanks